code/crates/core-driver/src/driver.rs

```diff
@@ -395,15 +438,39 @@ where
             return Ok(None);
         };
 
-        let round_input = self.multiplex_vote_threshold(output, vote_round);
+        if let VKOutput::PolkaValue(val) = &output {
+            self.store_polka_certificate(vote_round, val);
+        }
 
+        let round_input = self.multiplex_vote_threshold(output, vote_round);
         if round_input == RoundInput::NoInput {
             return Ok(None);
         }
 
         self.apply_input(vote_round, round_input)
     }
 
+    fn store_polka_certificate(&mut self, vote_round: Round, value_id: &ValueId<Ctx>) {
```

Comment:
Ref: https://github.com/informalsystems/malachite/blob/ad7350e81006dd28a1351c4eb33a2349208951cf/specs/consensus/quint/votekeeper.qnt#L541

Reply:
TODO: Check that the certificate we have constructed is valid.

Reply:
TODO: Rename this method to `get_polka_certificate`

Reply:
TODO: Move this method to the `VoteKeeper` and only store the certificate in `polka_certificates` here